apps/api/src/app/events/services/mail-service/handlers/base.handler.ts

```diff
@@ -0,0 +1,24 @@
+import { IEmailOptions, IEmailProvider } from '@notifire/core';
+import { ChannelTypeEnum } from '@notifire/shared';
+import { IMailHandler } from '../interfaces/send.handler.interface';
+
+export abstract class BaseHandler implements IMailHandler {
+  protected readonly providerId;
+  protected readonly channelType;
+  protected provider: IEmailProvider;
+
+  protected constructor(providerId: string, channelType: string) {
+    this.providerId = providerId;
+    this.channelType = channelType;
+  }
```

Comment:
```suggestion
  protected constructor(private providerId: string, private channelType: string) {
 
  }
```

In typescript you can use it like this and this will automatically create a private class member